### ISSUE TITLE
Container does not work on host with ipv6 disabled

### ISSUE THREAD

**Issue by glensc** [linuxserver/docker-swag#92]
_copied from https://github.com/linuxserver/docker-swag/issues/43#issuecomment-782581808_

```yml
# docker-compose.yml
version: "2.1"

services:
  swag:
    image: ghcr.io/linuxserver/swag
    container_name: swag
    cap_add:
      - NET_ADMIN
    environment:
      - PUID=501
      - PGID=501
      - TZ=Europe/Tallinn
      - URL=dyndns.example.org
      - SUBDOMAINS=
      - VALIDATION=http
      - CERTPROVIDER= #optional
      - DNSPLUGIN=cloudflare #optional
      - PROPAGATION= #optional
      - DUCKDNSTOKEN= #optional
      - EMAIL= #optional
      - ONLY_SUBDOMAINS=false #optional
      - EXTRA_DOMAINS= #optional
      - STAGING=false #optional
      - MAXMINDDB_LICENSE_KEY= #optional
    volumes:
      - ./config:/config
    ports:
      - 443:443
      - 80:80 #optional
    restart: unless-stopped

networks:
  default:
    name: swag

# vim:ts=2:sw=2:et
```

```
glen@acai:~/scm/swag$ docker inspect ghcr.io/linuxserver/swag -f '{{ json .ContainerConfig.Labels }}'  | jq
{
  "build_version": "Linuxserver.io version:- 1.12.0-ls42 Build-date:- 2021-02-18T05:35:06+00:00",
  "maintainer": "aptalca"
}
```

```
glen@acai:~/scm/swag$ rm -rf config

glen@acai:~/scm/swag$ docker-compose up -d
Starting swag ... done

glen@acai:~/scm/swag$ docker-compose logs --tail=20 -t
Attaching to swag                                                                                              
swag    | 2021-02-20T07:44:23.261374104Z [services.d] starting services
swag    | 2021-02-20T07:44:23.287080938Z [services.d] done.
swag    | 2021-02-20T07:44:23.413942329Z nginx: [emerg] socket() [::]:80 failed (97: Address family not support
ed by protocol)
swag    | 2021-02-20T07:44:24.013937160Z Server ready
swag    | 2021-02-20T07:44:24.382779503Z nginx: [emerg] socket() [::]:80 failed (97: Address family not support
ed by protocol)
swag    | 2021-02-20T07:44:25.441645760Z nginx: [emerg] socket() [::]:80 failed (97: Address family not support
ed by protocol)
swag    | 2021-02-20T07:44:26.449157126Z nginx: [emerg] socket() [::]:80 failed (97: Address family not support
ed by protocol)
swag    | 2021-02-20T07:44:27.449683444Z nginx: [emerg] socket() [::]:80 failed (97: Address family not support
ed by protocol)
swag    | 2021-02-20T07:44:28.445631179Z nginx: [emerg] socket() [::]:80 failed (97: Address family not support
ed by protocol)
```

and the service (nginx stays down), connection refused.

```
glen@acai:~/scm/swag$ curl localhost
curl: (56) Recv failure: Connection reset by peer


6 glen@acai:~/scm/swag$ telnet localhost 80
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
Connection closed by foreign host.
```

It's really connection refused from application point, but docker port forwarding makes error appear as connection reset.

```
glen@acai:~/scm/swag$ docker-compose exec swag netstat -nat|grep LISTEN
tcp        0      0 127.0.0.11:34147        0.0.0.0:*               LISTEN
tcp        0      0 127.0.0.1:9000          0.0.0.0:*               LISTEN
glen@acai:~/scm/swag$
```

ipv6 is disabled via kernel commandline here;

```
glen@acai:~/scm/swag$ cat /proc/cmdline
... ipv6.disable=1
```

**Comment by glensc** [comment]
_From https://github.com/linuxserver/docker-swag/issues/43#issuecomment-782584193_

So far, the fix for me is to remove ipv6 listen lines:


```diff
diff --git a/config/nginx/site-confs/default b/config/nginx/site-confs/default
index 6d76de6..980e5be 100644
--- a/config/nginx/site-confs/default
+++ b/config/nginx/site-confs/default
@@ -5,7 +5,6 @@ error_page 502 /502.html;
 # redirect all traffic to https
 server {
        listen 80 default_server;
-       listen [::]:80 default_server;
        server_name _;
        return 301 https://$host$request_uri;
 }
@@ -13,7 +12,6 @@ server {
 # main server block
 server {
        listen 443 ssl http2 default_server;
-       listen [::]:443 ssl http2 default_server;

        root /config/www;
        index index.html index.htm index.php;
(END)
```

**Comment by B-Han02** [comment]
Found this when googling for a solution to this issue.
Look like they added the option to disable IPv6 when starting the container.
Not a programmer or I would've fixed it myself.
https://github.com/jc21/nginx-proxy-manager/commit/bdb591af9e1aea81820c3fa11da9555f9672c653

**Comment by glensc** [comment]
@B-Han02 it's different product, it's layer on top of nginx, which generates nginx configuration.

But their solution there is exactly what I've indicated, drop lines with `listen ::`:

- https://github.com/jc21/nginx-proxy-manager/commit/bdb591af9e1aea81820c3fa11da9555f9672c653#diff-4105b7e7c8962737f7f0041242295d837ea31a4369299cf8513dc4be81b2e740

**Comment by j0nnymoe** [comment]
If the case is to remove all the ipv6 lines, then what about our users that are using IPv6?

**Comment by glensc** [comment]
@j0nnymoe I think for those users ipv4 will be used.

... or ipv4-over-v6 address translation. I don't think anybody has ipv6-only system or if they do, they would still have ipv4-over-v6 translation enabled. whoever runs ipv6-only, must be knowing what they are doing, so they are aware of risks of half internet being unreachable :). actually the internet clause probably not valid, as they would have ipv6 gateway which would do the translations for them.

so, tl;dr, ipv6 only systems need a transition mechanism to reach ipv4 addresses:
- https://en.wikipedia.org/wiki/IPv6#Transition_mechanisms

**Comment by glensc** [comment]
@j0nnymoe also, I just found, that you must do some extra steps to even make use of ipv6 in docker:
- https://docs.docker.com/config/daemon/ipv6/

so, looks like current users were using ipv4 anyway?

**Comment by glensc** [comment]
So, proceed making PR with these changes?
- https://github.com/linuxserver/docker-swag/issues/92#issuecomment-783237000

**Comment by Roxedus** [comment]
There's just a matter of time until someone wants to add them again. Doing some documentation going over how to enable/disable might be worth looking into, depending on which default we decide to go with.

**Comment by nemchik** [comment]
> So, proceed making PR with these changes?
> 
> * [#92 (comment)](https://github.com/linuxserver/docker-swag/issues/92#issuecomment-783237000)

We do have users who use ipv6 (we've explicitly had support requests for this). I would prefer not to remove them. The out of the box configuration for most common Linux distros is ipv6 isn't explicitly disabled (even if it's not configured to be used by the user). It makes more sense that we could add an init script to see if ipv6 is usable and if any ipv6 configurations aren't commented out and then warn the user if their configs need adjusting. I'm also not in favor of automatically making changes to user configs (ex: detect and replace). We could also add a comment line above each ipv6 line in the configs to make it more obvious what to look for.

**Comment by jeffbyrnes** [comment]
Took me a long while to figure out what was happening, and once I realized NGINX was getting stuck, and found this issue, the removal of IPv6 listeners was the fix.

Perhaps it’d be good to include something in the README?
